solidity/contracts/bridge/DepositorProxy.sol

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity 0.8.17;
+
+import {BTCUtils} from "@keep-network/bitcoin-spv-sol/contracts/BTCUtils.sol";
+
+import "./BitcoinTx.sol";
+import "./Bridge.sol";
+import "./Deposit.sol";
+import "./../vault/TBTCVault.sol";
+import "./../vault/TBTCOptimisticMinting.sol";
+import "./../token/TBTC.sol";
+
+// TODO: Make it safe for upgradeable contracts.
+// TODO: Document the contract.
+// TODO: Move to another directory?
+abstract contract DepositorProxy {
+    using BTCUtils for bytes;
+
+    /// @notice Multiplier to convert satoshi to TBTC token units.
+    uint256 public constant SATOSHI_MULTIPLIER = 10**10;
+
+    Bridge public bridge;
+    TBTCVault public tbtcVault;
+    TBTC public tbtcToken;
+
+    mapping(uint256 => bool) public pendingDeposits;
+
+    event DepositInitialized(
+        uint256 indexed depositKey,
+        uint32 initiatedAt
+    );
+
+    event DepositFinalized(
+        uint256 indexed depositKey,
+        uint256 tbtcAmount,
+        uint32 finalizedAt
+    );
+
+    constructor(address _bridge, address _tbtcVault, address _tbtcToken) {
+        bridge = Bridge(_bridge);
+        tbtcVault = TBTCVault(_tbtcVault);
+        tbtcToken = TBTC(_tbtcToken);
+    }
+
+    function initializeDeposit(
+        BitcoinTx.Info calldata fundingTx,
+        Deposit.DepositRevealInfo calldata reveal,
+        bytes32 extraData
+    ) internal {
```

Comment:
Can this function return the calculated `depositKey`?

Reply:
Of course! See fb2acde8